network.go

```diff
@@ -852,6 +852,10 @@ func (n *network) ipamAllocate() ([]func(), error) {
 				if ip = net.ParseIP(v); ip == nil {
 					return nil, types.BadRequestErrorf("non parsable secondary ip address %s (%s) passed for network %s", k, v, n.Name())
 				}
+				if !d.Pool.Contains(ip) {
```

Comment:
This is a valid use case: default gateway that does not belong to the fixed cidr